snuba/utils/streams/batching.py

```diff
@@ -114,19 +131,18 @@ class BatchingKafkaConsumer:
 
     def __init__(
         self,
-        topics: Union[str, Sequence[str]],
+        consumer: Consumer,
+        topic: Sequence[str],
         worker: AbstractBatchWorker,
         max_batch_size: int,
         max_batch_time: int,
-        bootstrap_servers: Sequence[str],
         group_id: str,
         metrics: MetricsBackend,
         producer: Optional[Producer] = None,
         commit_log_topic: Optional[str] = None,
-        auto_offset_reset: str = "error",
-        queued_max_messages_kbytes: int = DEFAULT_QUEUED_MAX_MESSAGE_KBYTES,
-        queued_min_messages: int = DEFAULT_QUEUED_MIN_MESSAGES,
     ) -> None:
```

Comment:
There are several changes that will be happening here in subsequent changes:

- `Consumer` (which is now a `confluent_kafka.Consumer`) instance will become a `snuba.utils.streams.abstract.Consumer` reference. There will be a `snuba.utils.streams.kafka.Consumer` class that wraps the Confluent Kafka consumer that satisfies that abstract interface.
- `group_id`, `producer` and `commit_log_topic` will be removed when a separate `abstract.Consumer` implementation is introduced that is responsible for publishing to the commit log — there is no reason for the batching consumer to be aware of that concern. This also removes an implicit requirement between those three parameters that isn't really defined anywhere except in assertions here to satisfy the type checker.

At the end of this, the constructor will be:

```python
    def __init__(
        self,
        consumer: Consumer,
        topic: Sequence[str],  # may move to `run` instead, but no rush
        worker: AbstractBatchWorker,
        max_batch_size: int,
        max_batch_time: int,
        metrics: MetricsBackend,
    ) -> None:
```